### ISSUE TITLE
IJulia / IPython / Jupyter combination not working

### ISSUE THREAD

**Issue by sax25** [JuliaLang/IJulia.jl#289]
Hallo

[using Linux/Ubuntu14.04 + Firefox 37 + Julia 0.3.7 + IJulia 0.2.1]

My Problem:
=============
I recently have upgraded from Python2.7 to Python3.4 and thereby (unknowingly) from IPython to Jupyter.
Since than not a single one of my IJulia notebooks is working anymore. (without throwing any error on console - and till now I haven't found any errorlog files for either julia nor pyhon)

Whats more: 
============
- The whole Jupyter seems to be in a kind of "dead" or passive mode. I can select files from the tree-view and create new files,  which both open but don't run.
- The "Run All" just skips through the complete notebook in no time and does nothing.
- It's also impossible to switch to different kernels from the Jupyter menu. 
  This throws an error to the Firefox Error Console: 

     TypeError: this.notebook.session.kernel is null

     Source: http://localhost:8998/static/notebook/js/kernelselector.js?v=20150402145919

     Line: 222

  (At least this is an error. It's the only one I get anywhere)

What I tried:
=============
- I agreed to convert old notebooks to the (seemingly mandantory) new format.
- I have trusted the notebooks.
- I have checked the config files and have a Julia kernel. 
- In Jupyter in the top right there is the Julia logo. Beneath is a grey dot with "Julia" beside it. Top left is the Jupyter logo. (I interpret this view as "You are using Jupyter with an Julia kernel")
- Same behaviour in Chromium

Does anyone encounter a similiar problem or has any ideas to solve this??

**Comment by stevengj** [comment]
Maybe similar to #286?   Try `Pkg.build("IJulia")`?

**Comment by sax25** [comment]
Yes, I tried that. Also had a look both with 

ipython notebook --profile julia --debug  

and

ijulia.jl edit "verbose=true"

- No error messages.
- Kernel started, when I opened a new or old file. 
- Kernel didn't throw any "restarting" messages. 
- Kernel shuts down, when I choose "Close and Halt" in Jupyter / opened notebookfile.

Just a warning from IPython:

- UserWarning: Setting kernel_cmd is deprecated, use kernel_spec to start different kernels.

And one from IJulia:

- [W 00:32:48.870 NotebookApp] 404 GET /kernelspecs/julia%200.3/logo-64x64.png (127.0.0.1) 39.24ms referer=http://localhost:8998/notebooks/Untitled3.ipynb

**Comment by sax25** [comment]
[SOLVED]

I fixed it! (https://github.com/ipython/ipython/issues/7943 helped a lot...)

Short version:

- manually ALLOW cookies for localhost via preferences (I had to do that, although localhost wasn't listet in firefox' cookie policies at all)
- ALLOW 3rd party cookies
- clear cache
- clear HTTP-auth
- drop session cookies
- drop domain cookies

And after that it works. 

From the now running Jupyter I saw that the grey dot besides the kernel name obviously meant "doesn't work". Because "working" is indicated by a grey circle.
Maybe that should be changed to a red vs. green dot...
